### PR TITLE
test:  backports knuu optimizations from 3493

### LINKED ISSUE #3480: Optimizing deployment process (With the new release candidate of knuu) for large-scale network tests
For the large-scale network test consisting of roughly 100 nodes, the current e2e tests require building an image for each node locally before deployment. This process is very time-consuming and slows down deployment as we need to build one image per node. knuu developers are already investigating this issue, and a new release  `v0.13.3.-rc.4` is on the way to make the process faster.

In an attempt to incorporate the first release candidate from that repository, a few errors have surfaced (for more details please see https://github.com/celestiaorg/celestia-app/pull/3456). In this issue, I aim to investigate these issues, resolve them, and ultimately integrate the new release candidate  `v0.13.3.-rc.4` into our network tests to enable a faster deployment process.

### PR DESCRIPTION
**Pull request by staheri14**
Closes #3480 
This PR backports many optimizations introduced in the big block tests via https://github.com/celestiaorg/celestia-app/pull/3493 to the `main` branch.

 Will be ready for review after merging https://github.com/celestiaorg/celestia-app/pull/3514

### CODE REVIEW

**Comment by staheri14** [comment]
Since the optimization work was done by @smuu, assigning him to the PR for comments/questions.

**Comment by smuu** [comment]
When this is merged, we can close https://github.com/celestiaorg/celestia-app/issues/3480

File: go.mod
Comment (staheri14) [inline go.mod:32]: @smuu Out of curiosity, why this change is needed? I backported it from the other PR https://github.com/celestiaorg/celestia-app/pull/3493/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R32

**Review by evan-forbes — APPROVED** [review]

File: test/e2e/testnet/test_helpers.go
Comment (rach-id) [inline test/e2e/testnet/test_helpers.go:87]: Can we include the review feedback listed in: https://github.com/celestiaorg/celestia-app/pull/3493#pullrequestreview-2064670299 before merging? since we're merging in main, it makes sense to clean this up.

File: test/e2e/testnet/node.go
Comment (rach-id) [inline test/e2e/testnet/node.go:304]: no more proxy instead of port forward? or that will come later?

**Review by rach-id — COMMENTED** [review]
left a comment to clean the implementation more before merging

File: test/e2e/testnet/node.go
Comment (cmwaters) [inline test/e2e/testnet/node.go:231]: So we are supposed to add the folder after committing now?

File: test/e2e/testnet/node.go
Comment (cmwaters) [reply test/e2e/testnet/node.go:304]: Yeah good point. What order should this PR be merged in respect to the proxying ports PR

File: test/e2e/testnet/test_helpers.go
Comment (cmwaters) [inline test/e2e/testnet/test_helpers.go:30]: Where does this get used?

**Comment by cmwaters** [comment]
Do you mind specifying what optimisations this makes to knuu?

**Comment by staheri14** [comment]
> When this is merged, we can close #3480

@smuu Thanks! Added to the PR description.

File: test/e2e/testnet/node.go
Comment (smuu) [reply test/e2e/testnet/node.go:231]: The instances do not differ except for the files we add there.
- If we add the files before committing, it will build a container for each Instance, which takes very long if we want to start 100 validators
- If we add the files after committing, knuu can use a cached version of the Instance, which speeds up if you want to start many nodes that use the same base, but different files.

File: test/e2e/testnet/test_helpers.go
Comment (smuu) [reply test/e2e/testnet/test_helpers.go:30]: can be removed when https://github.com/celestiaorg/celestia-app/pull/3505 is merged

**Comment by evan-forbes** [comment]
blocked by #3505

**Comment by staheri14** [comment]
> blocked by #3505

#3505 has been merged, and I have revised the content of this PR accordingly. cc:@smuu

**Comment by smuu** [comment]
Let me know if you have any other questions about these optimizations.

File: test/e2e/testnet/node.go
Comment (evan-forbes) [reply test/e2e/testnet/node.go:231]: what's the best way to keep track of important details such as this @smuu? we could add a code comment above, but I imagine this would also be important to know when creating tests for node as well

**Review by evan-forbes — APPROVED** [review]

**Comment by staheri14** [comment]
[To reviewers] I run all the e2e tests and benchmark tests and can confirm that they all work fine.

File: test/e2e/testnet/node.go
Comment (cmwaters) [inline test/e2e/testnet/node.go:319]: This can be done in a follow up but this function should really be two different functions since it performs two different actions.

In fact, I would be more in favour of grouping `StartAsync` and `WaitInstanceIsRunning` together

File: test/e2e/testnet/test_helpers.go
Comment (cmwaters) [inline test/e2e/testnet/test_helpers.go:22]: Q: What is this used by? I don't see any code in this PR that links to `JSONRPCError`

**Review by cmwaters — APPROVED** [review]

File: test/e2e/testnet/node.go
Comment (staheri14) [reply test/e2e/testnet/node.go:319]: Tracking issue https://github.com/celestiaorg/celestia-app/issues/3599

File: test/e2e/testnet/test_helpers.go
Comment (staheri14) [reply test/e2e/testnet/test_helpers.go:22]: You are right, I also think it is safe to remove it as I couldn't also find any usage for it anywhere else.

File: test/e2e/testnet/node.go
Comment (staheri14) [reply test/e2e/testnet/node.go:231]: I've created a tracking issue for this https://github.com/celestiaorg/celestia-app/issues/3601

File: test/e2e/testnet/test_helpers.go
Comment (staheri14) [reply test/e2e/testnet/test_helpers.go:22]: Tracking issue #3600

**Pull request merged**
